packages/governance/src/paramGovernance/paramManager.js

```diff
@@ -0,0 +1,301 @@
+    };
+
+    const publicMethods = Far(`Parameter ${name}`, {
+      getValue: () => currentAmount,
+      getInternalValue: () => currentInvitation,
+      assertType: assertInvitation,
+      makeDescription,
+      makeShortDescription,
+      getType: () => ParamType.INVITATION,
+    });
+
+    // CRUCIAL: here we're creating the update functions that can change the
+    // values of the governed contract's parameters. We'll return the updateFns
+    // to our caller. They must handle them carefully to ensure that they end up
+    // in appropriate hands.
+    updateFns[`update${name}`] = setInvitation;
+    namesToParams.init(name, publicMethods);
+    return name;
+  };
+
+  /** @type {(name: string, value: Invitation, builder: ParamManagerBuilder) => Promise<ParamManagerBuilder>} */
+  const addInvitation = async (name, value, builder) => {
+    assertKeywordName(name);
+    await Promise.all([
+      assertInvitation(value),
+      buildInvitationParam(name, value),
+    ]);
+
+    return builder;
+  };
+
+  // PARAM MANAGER METHODS ////////////////////////////////////////////////////
+
+  const getParam = name => {
+    return namesToParams.get(name).makeDescription();
+  };
+
+  const getTypedParam = (type, name) => {
+    const param = namesToParams.get(name);
+    assert(type === param.getType(), X`${name} is not ${type}`);
+    return param.getValue();
+  };
+
+  const getParamList = () => {
+    return harden(namesToParams.keys().map(k => getParam(k)));
+  };
+
+  // should be exposed within contracts, and not externally, for invitations
+  const getInternalParamValue = name => {
+    return namesToParams.get(name).getInternalValue();
+  };
+
+  const getParams = () => {
+    /** @type {Record<Keyword,ParamDescription>} */
+    const descriptions = {};
+    namesToParams.entries().forEach(([name, param]) => {
+      descriptions[name] = param.makeShortDescription();
+    });
+    return harden(descriptions);
+  };
+
+  const makeParamManager = updateFunctions => {
+    // CRUCIAL: Contracts that call buildParamManager should only export the
+    // resulting paramManager to their creatorFacet, where it will be picked up by
+    // contractGovernor. The getParams method can be shared widely.
+    return Far('param manager', {
+      getParams,
+      getSubscription: () => subscription,
+      getAmount: name => getTypedParam(ParamType.AMOUNT, name),
+      getBrand: name => getTypedParam(ParamType.BRAND, name),
+      getInstance: name => getTypedParam(ParamType.INSTANCE, name),
+      getInstallation: name => getTypedParam(ParamType.INSTALLATION, name),
+      getInvitationAmount: name => getTypedParam(ParamType.INVITATION, name),
+      getNat: name => getTypedParam(ParamType.NAT, name),
+      getRatio: name => getTypedParam(ParamType.RATIO, name),
+      getString: name => getTypedParam(ParamType.STRING, name),
+      getUnknown: name => getTypedParam(ParamType.UNKNOWN, name),
```

Comment:
It seems like it would be particularly handy to have return type annotations on these.

Reply:
Ah... I see... the `ParamManagerBuilder` declaration below has the relevant effect.